### ISSUE TITLE
💅 useGenericFontNames triggers in the @supports query

### ISSUE THREAD

**Issue by ivany4** [biomejs/biome#8845]
### Environment information

Have a media query in CSS:

```css
@supports (font: -apple-system-body) {
  /* ... */
}
```

Triggers `lint/a11y/useGenericFontNames: Generic font family missing.`

Even if it's a non-standard apple-specific token, the error message and the choice of the rule to trigger is misleading.

### Rule name

lint/a11y/useGenericFontNames

### Playground link

https://biomejs.dev/playground/?code=QABzAHUAcABwAG8AcgB0AHMAIAAoAGYAbwBuAHQAOgAgAC0AYQBwAHAAbABlAC0AcwB5AHMAdABlAG0ALQBiAG8AZAB5ACkAIAB7AAoAIAAgAC8AKgAgAC4ALgAuACAAKgAvAAoAfQA%3D&language=css

### Expected result

Probably best would be to have the message about unsupported/non-standard name (coming from somewhere other than a11y rule) and the ability to exclude this token in the configuration.

Or no error at all, when used inside `@supports`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
